plot_radial_build.py

```diff
@@ -0,0 +1,150 @@
+import matplotlib.colors
+import yaml
+import argparse
+
+def build_composition_string(composition, max_characters):
+    """
+    Assembles string from composition dict for use in radial build plot
+
+    Arguments:
+        composition (dict): "material name (str)":volume_fraction (float)
+
+    Returns:
+        comp_string (string): formatted string with material definition
+    """
+
+    comp_string = ''
+    for material, fraction in composition.items():
+        
+        mat_string = f'{material}: {round(fraction*100,3)}%, '
+        line_len =len(comp_string)-(comp_string+mat_string).rfind('\n')
+
+        if line_len > max_characters:
+            comp_string += '\n' + mat_string
+        else:
+            comp_string += mat_string
+
+    return comp_string[0:-2]
+
+def plot_radial_build(build, title, colors = None, 
+                      max_characters = 35, max_thickness = None, size = (8,4), 
+                      unit = 'cm'):
+    """
+    Creates a radial build plot, with layers scaled between a minimum and
+        maximum pixel width to preserve readability
+    
+    Arguments:
+        build (dict): {"layer name": {"thickness": (float),
+                                      "composition": {
+                                        "material name": fraction (float)
+                                }
+                            }
+                        }
+        title (string): title for plot and filename to save to
+        colors (list of str): list of matplotlib color strings. 
+            If specific colors are desired for each layer they can be added here
+        max_characters (float): maximum length of a line before wrapping the 
+            text
+        max_thickness (float): maximum thickness of layer to display, useful
+            for reducing the total size of the figure.
+        size (iter of float): figure size, inches. (width, height)
+        unit (str): Unit of thickness values
+    """
+
+    char_to_height = 1.15
+    min_line_height = 8
+    min_lines = 2
+    height = char_to_height*max_characters
+
+    if colors is None: 
+        colors = list(matplotlib.colors.XKCD_COLORS.values())[0:len(build)]
+
+    #initialize list for lower left corner of each layer rectangle
+    ll = [0,0]
+    plt.figure(1, figsize=size)
+    plt.tight_layout()
+    ax = plt.gca()
+    ax.set_ylim(0,height+1)
+
+    total_thickness = 0
+    for layer, color in zip(build, colors):
+
+        comp_string = build_composition_string(build[layer]['composition'],
+                                               max_characters)
+        newlines = comp_string.count('\n')
+
+        # adjust thicknesses
+        if max_thickness is not None:
+            build[layer]['thickness'] = min(max_thickness,
+                                            build[layer]['thickness'])
+            
+        thickness = max(build[layer]['thickness'], 
+                        (min_lines + newlines)*min_line_height)
```

Comment:
If `max_thickness` is always defined (default = something very large), then this can all be simplified as:
```suggestion
        newlines = comp_string.count('\n')
        min_thickness = (min_lines + newlines) * min_line_height
        
        # adjust thicknesses
        thickness = min(max(build[layer]['thickness'], min_thickness), max_thickness)

```